mat64/pool.go

```diff
@@ -0,0 +1,52 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import (
+	"sync"
+
+	"github.com/gonum/blas/blas64"
+)
+
+// pool contains size stratified workspace Dense pools.
+// Each pool element i returns sized matrices with a data
+// slice capped at 1<<i.
+var pool [63]sync.Pool
+
+func init() {
+	for i := range pool {
+		l := 1 << uint(i)
+		pool[i].New = func() interface{} {
+			return &Dense{mat: blas64.General{
+				Data: make([]float64, l),
+			}}
+		}
+	}
+}
+
+// getWorkspace returns a *Dense of size r×c and a data slice
+// with a cap that is less than 2*r*c. If clear is true, the
+// data slice visible through the Matrix interface is zeroed.
```

Comment:
Since this is not exported, people will only read the comment when the code is in front of them, where they can see that this is the case. The comment here is to explain the intent - that the only zeroing is on elements that can be seen when only the Matrix methods are used. If it were exported, I would have both.